src/frequenz/sdk/data_ingestion/resampling/timeseries_resampling_actor.py

```diff
@@ -0,0 +1,159 @@
+
+
+@actor
+class TimeSeriesResamplingActor:
+    """TimeSeriesResamplingActor used to ingest component data and resample it."""
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        data_provider: DataProvider,
+        subscription_receiver: Receiver[SubscriptionRequest],
+        period_s: float = 0.2,
+        max_data_age_in_periods: float = 3.0,
+        resampling_function: ResamplingFunction = average,
+    ) -> None:
+        """Initialize the TimeSeriesResamplingActor.
+
+        Args:
+            data_provider: class implementing the DataProvider interface for
+                getting raw component data
+            subscription_receiver: channel for receiving subscription requests
+            period_s: resampling frequency, in seconds
+            max_data_age_in_periods: max age that samples shouldn't exceed in order
+                to be used in the resampling function
+            resampling_function: function to be applied to a sequence of samples within
+                a resampling period to produce a single output sample
+        """
+        self._data_provider = data_provider
+        self._subscription_receiver = subscription_receiver
+        self._period_s = period_s
+        self._max_data_age_in_periods: float = max_data_age_in_periods
+        self._resampling_function: ResamplingFunction = resampling_function
+
+        self._resampler = TimeSeriesResampler(
+            period_s=period_s,
+            max_data_age_in_periods=max_data_age_in_periods,
+            resampling_function=resampling_function,
+        )
+
+        # `TimeSeriesId`s as dictionary keys will likely have to change to
+        # `Subscription`s, which take `start_time` into account for reading
+        # historical data
+        self._input_receivers: Dict[TimeSeriesId, Receiver[Optional[Sample]]] = {}
+        self._output_channels: Dict[TimeSeriesId, Broadcast[Optional[Sample]]] = {}
+        self._resampling_timer = Timer(interval=self._period_s)
+
+    def subscribe(self, subscription: Subscription) -> Receiver[Optional[Sample]]:
+        """Subscribe for data for a specific time series.
+
+        Args:
+            subscription: subscription consisting of `time_series_id`, which is made up
+                of a (component_id, metric_id) pair and `start_time`, which is the
+                point in time from which to subscribe  for data or `None` to subscribe
+                for data from the live component data stream
+
+        Returns:
+            receiver for receiving samples for a specific time series id
+        """
+        if subscription.time_series_id not in self._output_channels:
+            response: SubscriptionResponse = self._data_provider.subscribe(
+                time_series_id=subscription.time_series_id,
+                # Ignore parquet for now [later => start_time=subscription.start_time]
+                start_time=None,
+            )
```

Comment:
You should ignore start_time in Subscription class. Not here.
There are methods like post_init, to do so.

> Ignore parquet for now [later => start_time=subscription.start_time]

You should also describe it in Subscription class. Not here. Please change to `Ignore historical data for now. In result user would only data from current timestamp.`

Reply:
This will most likely change (no subscription response and no `start_time` in the request), so let's leave it for now.